implementation/src/main/java/io/smallrye/metrics/legacyapi/LegacyMetricRegistryAdapter.java

```diff
@@ -465,7 +463,8 @@ <T> FunctionCounterAdapter<T> internalCounter(MpMetadata metadata, T obj, ToDoub
 
         FunctionCounterAdapter<T> result = checkCast(FunctionCounterAdapter.class, metadata,
                 constructedMeters.computeIfAbsent(id, k -> new FunctionCounterAdapter(obj, func)));
-        addNameToApplicationMap(id.toMetricID());
+        if (isAppnameResolverPresent)
+            addNameToApplicationMap(id.toMetricID());
```

Comment:
would be cleaner to have addNameToApplicationMap take a MetricDescriptor as a parameter, and do the isAppnameResolverPresent check inside addNameToApplicationMap (before it calls id.toMetricID())